runtime/queries/markdown_inline/highlights.scm

```diff
@@ -33,6 +33,11 @@
   ] @markup.link
   (#set! conceal ""))
 
+(inline_link
+  (link_text) @markup.link.label
+  (link_destination) @markup.link
+  (#set! @markup.link.label "url" @markup.link))
```

Comment:
How does this interact with the conceal later? (What happens if `conceallevel` is set?)

Reply:
It works beautifully https://github.com/neovim/neovim/pull/27109#issuecomment-1902244611